### ISSUE TITLE
Segmentation fault (core dumped) on Hyprland using wlr/workspaces module with persistent worspaces enabled

### ISSUE THREAD

**Issue by pyunramura** [Alexays/Waybar#2418]
I'm running into some odd behavior when selecting an inactive persistent workspace using the `wlr/workspaces` module with Waybar and Hyprland. Details below.

Behavior:  
The core dump from above will only happen when clicking on an inactive persistent workspace,. Clicking on an active one or inactive one works as expected. Scrolling behavior works as well.

I've tried running Waybar with `--log-level trace` to get more useful output, but `Segmentation fault (core dumped)` is unfortunately the only relevant line from the logs.

I've also tried changing the `activate` trigger between `on-click` `on-click-right` and `on-click-middle`, with no effect.

Version info:  
OS: NixOS Unstable  
Hyprland: Built from commit [8e04a80](https://github.com/hyprwm/Hyprland/commit/8e04a80e60983f5def26bdcaea701040fea9a7ae).   
Waybar: v0.9.20 Using `-Dexperimental=true` override as referenced from #2116. 

I'm using the `wlr/workspaces` module currently as the `hyprland/workspaces` module wasn't working fully when I tied it last.

Relevant config snippet:
```json
"wlr/workspaces": {
    "all-outputs": false,
    "on-click": "activate",
    "on-scroll-up": "hyprctl dispatch workspace e+1",
    "on-scroll-down": "hyprctl dispatch workspace e-1",
    "sort-by-number": true,
    "format": "{icon}",
    "format-icons": {
        "1": "1",
        "2": "2",
        "3": "3",
        "4": "4",
        "5": "5",
        "6": "6",
        "7": "7",
        "8": "8",
        "9": "9",
        "10": "0"
    },
    "persistent_workspaces": {
        "1": [],
        "2": [],
        "3": [],
        "4": [],
        "5": [],
        "6": [],
        "7": [],
        "8": [],
        "9": [],
        "10": []
    }
}
```
If there's any other information I can provide, please let me know. I haven't seen another issue addressing the specific behavior I'm seeing, so hopefully it has already been solved and I'm just running an unfortunate mix of config options and package versions. If there's any insight into the problem I'm seeing though I'd be very thankful to have it.

**Comment by zjeffer** [comment]
Are you using the [hyprctl patch](https://aur.archlinux.org/cgit/aur.git/tree/hyprctl.patch?h=waybar-hyprland-git)?

**Comment by pyunramura** [comment]
Thank you for the quick reply. Now that I'm looking at it again, it seems that there are both the `waybar` and `waybar-hyprland` packages available for NixOS and I mistakenly installed the former on my system, and it looks like the latter already has the override I mentioned in my previous post. Both of these packages are on v0.9.21 currently.

I've just updated the `waybar-hyprland` paackage to v0.9.21 and the segfault behavior with `wlr/workspaces` has been resolved, fantastic news! I wasn't able to replicate my current persistent workspace behavior with the `hyprland/workspaces` module however. I've tried to replicate it using the persistent-workspaces configuration in my previous post, and using your snippet. Both with no luck. My original issue of  the `wlr/workspaces` segfault has been resolved though and it is perfectly usable in its current state, so closing this ticket now. Thanks so much for the advice.
